kbase-extension/static/kbase/js/widgets/narrative_core/upload/fileUploadWidget.js

```diff
@@ -112,6 +121,15 @@ define([
                     $dropzoneElem.find('#global-info').css({'display': 'none'});
                     $($dropzoneElem.find('#total-progress .progress-bar')).css({'width': '0'});
                 })
+                .on('canceled', (file) => {
+                    Promise.resolve(this.stagingServiceClient.delete({
+                        path: file.fullPath ? file.fullPath : file.name
```

Comment:
Why do we fallback to the filename if the full path is not available? Is it possible to delete a file with the same name in the root directory by accident this way?

Reply:
The full path is only available when a file is contained in a folder i.e. `testFolder/2.json` is a full path name. If a file isn't contained in a folder, then there is not a full path name defined, and just the file name is used i.e. `2.json`.

Reply:
Should this be popped out to a separate function, then we can call it on the 'error' event? I'm not sure if there's cases where partial files happen when there's an error.